desktop/core/src/desktop/lib/raz/clients_test.py

```diff
@@ -58,3 +63,39 @@ def test_check_acccess_s3_list_file(self):
 
 
   def test_check_acccess_s3_list_file_no_access(self): pass
+
+class AdlsRazClientTest(unittest.TestCase):
+
+  @classmethod
+  def setUpClass(cls):
+    if not RAZ.IS_ENABLED.get():
+      raise SkipTest
+  
+  def setUp(self):
+    self.username = 'csso_hueuser'
+  
+  def test_check_rename_operation(self):
+    with patch('desktop.lib.raz.raz_client.RazToken.get_delegation_token') as raz_token:
+      with patch('desktop.lib.raz.raz_client.requests.post') as requests_post:
+        with patch('desktop.lib.raz.raz_client.uuid.uuid4') as uuid:
+          with patch('desktop.lib.raz.raz_client.RazClient.check_access') as check_access:
+
+            reset = RAZ.API_URL.set_for_testing('https://raz_url:8000')
+            check_access.return_value = {'token': 'some_random_sas_token'}
+
+            try:
+              sas_token = AdlsRazClient(
+                username=self.username,
+                base_url='https://gethuestorage.dfs.core.windows.net'
+              ).get_url(
+                action='PUT',
+                path='https://gethuestorage.dfs.core.windows.net/data/user/csso_harsh.gupta/renamed_dir',
+                headers={'x-ms-version': '2019-12-12', 'x-ms-rename-source': '/data/user/csso_harsh.gupta/test_dir'})
+
+              check_access.assert_called_with(
+                headers={'x-ms-version': '2019-12-12', 'x-ms-rename-source': '/data/user/csso_harsh.gupta/test_dir?some_random_sas_token'},
```

Comment:
Nice!

However we don't have the base url in the updated destination dir?

Reply:
We have the base_url in the renamed dir (which is destination) and we dont need base_url in the header source path. Will improve the naming of these path too.